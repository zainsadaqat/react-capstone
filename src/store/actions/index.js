import axios from 'axios';

// Action Type
export const ENGLAND_DATA = 'ReactReduxCapstone/UK_BANK_HOLIDAYS/ENGLAND_DATA';
export const IRELAND_DATA = 'ReactReduxCapstone/UK_BANK_HOLIDAYS/IRELAND_DATA';
export const SCOTLAND_DATA =
  'ReactReduxCapstone/UK_BANK_HOLIDAYS/SCOTLAND_DATA';

export const initialState = {
  england: [],
  ireland: [],
  scotland: [],
};

// Action Creators
const englandData = [];
export const fetchEnglandData = () => async (dispatch) => {
  const response = await axios.get('https://www.gov.uk/bank-holidays.json');
  const englandData = await response.data;

  dispatch({
    type: ENGLAND_DATA,
    payload: englandData['england-and-wales'].events,
  });
};

export const fetchIrelandData = () => async (dispatch) => {
  const response = await axios.get('https://www.gov.uk/bank-holidays.json');
  const data = await response.data;

  dispatch({ type: IRELAND_DATA, payload: data['northern-ireland'].events });
};

export const fetchScotlandData = () => async (dispatch) => {
  const response = await axios.get('https://www.gov.uk/bank-holidays.json');
  const data = await response.data;

  dispatch({ type: SCOTLAND_DATA, payload: data.scotland.events });
};
