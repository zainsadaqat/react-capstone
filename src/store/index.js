import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import Reducer from './reducers';

const reducer = combineReducers({ Reducer });

const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
