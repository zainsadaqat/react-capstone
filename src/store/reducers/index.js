import {
  initialState,
  ENGLAND_DATA,
  IRELAND_DATA,
  SCOTLAND_DATA,
  ENGLAND_DATA_LENGTH,
  IRELAND_DATA_LENGTH,
  SCOTLAND_DATA_LENGTH,
} from '../actions';

const Reducer = (state = initialState, action) => {
  switch (action.type) {
    case ENGLAND_DATA:
      return { ...state, england: action.payload };
    case ENGLAND_DATA_LENGTH:
      return { ...state, englandDataLength: action.payload };
    case IRELAND_DATA:
      return { ...state, ireland: action.payload };
    case IRELAND_DATA_LENGTH:
      return { ...state, irelandDataLength: action.payload };
    case SCOTLAND_DATA:
      return { ...state, scotland: action.payload };
    case SCOTLAND_DATA_LENGTH:
      return { ...state, scotlandDataLength: action.payload };
    default:
      return state;
  }
};

export default Reducer;
