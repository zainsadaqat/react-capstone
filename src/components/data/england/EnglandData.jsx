import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchEnglandData } from '../../../store/actions/index';
import RightArrow from '../../../assets/icons/right-arrow-circle-icon.svg';

const EnglandData = () => {
  const dispatch = useDispatch();
  const england = useSelector((state) => state.Reducer.england);

  useEffect(() => {
    dispatch(fetchEnglandData());
  }, []);

  return (
    <div>
      <h2 className="country-name">England Data ({england.length})</h2>
      <ol className="card">
        {england.map((data) => (
          <li key={data.id} className="card-items">
            <span className="title">{data.title}</span>
            <span className="sub-portion">
              <span className="date">{data.date}</span>
              <span>
                <img src={RightArrow} alt="right arrow" className="icon" />
              </span>
            </span>
          </li>
        ))}
      </ol>
    </div>
  );
};

export default EnglandData;
