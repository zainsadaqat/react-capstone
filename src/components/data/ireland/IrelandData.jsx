import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchIrelandData } from '../../../store/actions/index';
import RightArrow from '../../../assets/icons/right-arrow-circle-icon.svg';

const IrelandData = ({ ireland }) => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchIrelandData());
  }, []);

  return (
    <div>
      <h2 className="country-name">Ireland Data</h2>
      <ol className="card">
        {ireland.map((data) => (
          <li key={data.id} className="card-items">
            <span className="title">{data.title}</span>
            <span className="sub-portion">
              <span className="date">{data.date}</span>
              <span>
                <img src={RightArrow} alt="right arrow" className="icon" />
              </span>
            </span>
          </li>
        ))}
      </ol>
    </div>
  );
};

export default IrelandData;

IrelandData.propTypes = {
  ireland: PropTypes.arrayOf(PropTypes.string).isRequired,
};
