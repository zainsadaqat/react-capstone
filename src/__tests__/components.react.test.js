import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from '../store/index';
import EnglandData from '../components/data/england/EnglandData';
import IrelandData from '../components/data/ireland/IrelandData';

describe('Testing react-redux-capstone project Components', () => {
  test('Testing EnglandData Component', () => {
    const englandData = renderer.create(
      <Provider store={store}>
        <EnglandData />
      </Provider>,
    );
    const tree = englandData.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Testing IrelandData Component', () => {
    const irelandData = renderer.create(
      <Provider store={store}>
        <IrelandData />
      </Provider>,
    );
    const tree = irelandData.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
