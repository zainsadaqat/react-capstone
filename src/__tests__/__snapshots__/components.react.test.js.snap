// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing react-redux-capstone project Components Testing EnglandData Component 1`] = `
<div>
  <h2
    className="country-name"
  >
    England Data (
    0
    )
  </h2>
  <ol
    className="card"
  />
</div>
`;

exports[`Testing react-redux-capstone project Components Testing IrelandData Component 1`] = `
<div>
  <h2
    className="country-name"
  >
    Ireland Data (
    0
    )
  </h2>
  <ol
    className="card"
  />
</div>
`;
